### PR TITLE
feat(frontend): migrate to TinyMCE 5

### PR DESCRIPTION
**Pull request by raphj**
#5409 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend/dist/index.mak [hunk 1]
```diff
@@ -133,15 +133,16 @@
...
     <script type="text/javascript" src="/assets/bootstrap/bootstrap-4.0.0-beta.2.js?token=${cache_token}" nonce="${csp_nonce}"></script>
 
-    <script type="text/javascript" src="/assets/tinymce-4.7.13/js/tinymce/jquery.tinymce.min.js?token=${cache_token}" nonce="${csp_nonce}"></script>
-    <script type="text/javascript" src="/assets/tinymce-4.7.13/js/tinymce/tinymce.min.js?token=${cache_token}" nonce="${csp_nonce}"></script>
-    <script type="text/javascript" src="/assets/tinymce-4.7.13/js/tinymce/themes/modern/theme.min.js?token=${cache_token}" nonce="${csp_nonce}"></script>
-
-    % for plugin in glob("assets/tinymce-4.7.13/js/tinymce/plugins/*/plugin.min.js"):
+    <script type="text/javascript" src="/assets/tinymce-5.10.3/js/tinymce/jquery.tinymce.min.js?token=${cache_token}" nonce="${csp_nonce}"></script>
+    <script type="text/javascript" src="/assets/tinymce-5.10.3/js/tinymce/tinymce.min.js?token=${cache_token}" nonce="${csp_nonce}"></script>
+    <!--<script type="text/javascript" src="/assets/tinymce-5.10.3/js/tinymce/themes/modern/theme.min.js?token=${cache_token}" nonce="${csp_nonce}"></script>-->
```
Comment (gbonaspetti) [inline frontend/dist/index.mak]: Forgotten comment?

File: frontend/src/css/Generic.styl [hunk 2]
```diff
@@ -35,28 +35,31 @@ a
...
-/*Editor footer*/
-div.mce-statusbar
-  flex 0 1 31px
+  top headerHeight
+
+// NOTE - RJ - 2022-02-14 - remove some borders on TinyMCE toolbars
+.tox-toolbar
+  background-image none !important // native TinyMCE borders are drawn using a background image
+  &__group
+    border: 0 !important
```
Comment (gbonaspetti) [inline frontend/src/css/Generic.styl]:
```suggestion
    border 0 !important
```

File: frontend/src/css/Generic.styl [hunk 3]
```diff
@@ -35,28 +35,31 @@ a
...
+    padding 0 !important
+
+    &:not(:last-child)
+      > :last-child
+        padding-right: 3px
```
Comment (gbonaspetti) [inline frontend/src/css/Generic.styl]:
```suggestion
        padding-right 3px
```

File: frontend_lib/webpack.standalone.config.js [hunk 4]
```diff
@@ -49,7 +49,7 @@ module.exports = {
   // they cause a lot of warnings in the browser's console
   module: {
     rules: [
-      isProduction ? {} : {
+      true || isProduction ? {} : {
```
Comment (gbonaspetti) [inline frontend_lib/webpack.standalone.config.js]:
```suggestion
      isProduction ? {} : {
```

File: frontend_app_workspace_advanced/src/css/index.styl [hunk 5]
```diff
@@ -25,7 +25,10 @@
   &__description
     &__text
       padding standardSpacing
+      position relative
+      height 13em
       &__textarea
+        height 13em
```
Comment (gbonaspetti) [inline frontend_app_workspace_advanced/src/css/index.styl]: Why `em`?

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: frontend/dist/assets/tinymce-5.10.3/test.html [hunk 6]
```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html>
+<html lang="fr">
+  <head>
+    <meta charset="utf-8"/>
+    <title>test tinymce</title>
```
[the remark this answers is not in the document]
Comment (raphj) [reply frontend/dist/assets/tinymce-5.10.3/test.html]: yes indeed, I'll remove it, thanks. good catch!

File: frontend/dist/index.mak (hunk 1 above)
Comment (raphj) [reply frontend/dist/index.mak]: yep, I'll remove it

File: frontend/src/css/Generic.styl (hunk 3 above)
Comment (raphj) [reply frontend/src/css/Generic.styl]: fixed

File: frontend/src/css/Generic.styl (hunk 2 above)
Comment (raphj) [reply frontend/src/css/Generic.styl]: fixed

File: frontend_lib/webpack.standalone.config.js (hunk 4 above)
Comment (raphj) [reply frontend_lib/webpack.standalone.config.js]: whoops, fixed.

File: frontend_app_workspace_advanced/src/css/index.styl (hunk 5 above)
Comment (raphj) [reply frontend_app_workspace_advanced/src/css/index.styl]: Easier to reason about, I changed to 200px.

**Review by gbonaspetti — APPROVED** [review]

**Review by Haskaris — APPROVED** [review]

File: frontend/src/css/Generic.styl [hunk 7]
```diff
@@ -1,4 +1,4 @@
-a
+
   &:hover, &:link, &:visited
     text-decoration none
```
Comment (gbonaspetti) [inline frontend/src/css/Generic.styl]: I think you deleted the `a` unintentionally

File: frontend_lib/src/component/Timeline/CommentArea.jsx [hunk 8]
```diff
@@ -313,13 +324,14 @@ export class CommentArea extends React.Component {
         <div
           className={classnames(
             `${props.customClass}__texteditor__textinput`,
-            'commentArea__textinput'
+            'commentArea__textinput',
+            props.wysiwyg && 'richtextedition'
```
Comment (gbonaspetti) [inline frontend_lib/src/component/Timeline/CommentArea.jsx]:
```suggestion
            { richtextedition: props.wysiwyg }
```

File: frontend_lib/src/helper.js [hunk 9]
```diff
@@ -857,6 +857,18 @@ const seekUsernameEnd = (text, offset) => {
   return offset
 }
 
+export const tinymceRemove = (selector) => {
+  try {
+    globalThis.tinymce.remove(selector)
+  } catch (e) {
+    if (e instanceof TypeError) {
+      console.error('HACK: removing TinyMCE raised a TypeError exception. If the message looks like "Can\'t access dead object", this probably means TinyMCE has been removed too late, or several times. Ignoring the exception but please fix this.', e)
```
Comment (gbonaspetti) [inline frontend_lib/src/helper.js]: It will be nice to create an issue for this

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: frontend_lib/src/helper.js (hunk 9 above)
Comment (raphj) [reply frontend_lib/src/helper.js]: done :-)

**Comment by raphj** [comment]
I'm merging this despise this being a big dev and the tests not having run so we can test this intensively tomorrow.

**Pull request merged**